demos/tests/args.py

```diff
@@ -106,3 +108,14 @@ def __init__(self, file_name):
 
     def resolve(self, context):
         return str(context.source_dir / self.file_name)
+
+
+class DemoOutputFileArg:
+    def __init__(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            self.out_dir = Path(temp_dir) / 'demo_output'
+        if not self.out_dir.is_dir():
+            self.out_dir.mkdir(parents=True)
+
+    def resolve(self, context):
+        return str(self.out_dir / 'output.avi')
```

Comment:
```suggestion
        return str(self.out_dir / '%09d.png')
```
It looks that the values for similarity metrics are different for you and CI. May be saving demo output as uncompressed images will eliminate the issue.